test/notification_cleaner_test/notification_cleaner_test.go

```diff
@@ -0,0 +1,139 @@
+	var ctx context.Context
+	var deleteInterceptor *storagefakes.FakeDeleteInterceptor
+	var deleteInterceptorProvider *storagefakes.FakeDeleteInterceptorProvider
+	var continueDelete chan struct{}
+	var deleteFinished chan struct{}
+
+	randomNotification := func() *types.Notification {
+		idBytes, err := uuid.NewV4()
+		Expect(err).ToNot(HaveOccurred())
+		return &types.Notification{
+			Base: types.Base{
+				ID:        idBytes.String(),
+				CreatedAt: time.Now(),
+				UpdatedAt: time.Now(),
+			},
+			Resource: "resource",
+			Type:     "CREATED",
+		}
+	}
+
+	getNotification := func(id string) {
+		obj, err := repository.Get(ctx, types.NotificationType, id)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(obj.GetID()).To(Equal(id))
+	}
+
+	BeforeSuite(func() {
+		ctx = context.Background()
+		deleteInterceptor = &storagefakes.FakeDeleteInterceptor{}
+		deleteInterceptor.OnTxDeleteStub = func(h storage.InterceptDeleteOnTxFunc) storage.InterceptDeleteOnTxFunc {
+			return h
+		}
+		continueDelete = make(chan struct{})
+		deleteFinished = make(chan struct{})
+		deleteInterceptor.AroundTxDeleteStub = func(h storage.InterceptDeleteAroundTxFunc) storage.InterceptDeleteAroundTxFunc {
+			return func(ctx context.Context, deletionCriteria ...query.Criterion) (types.ObjectList, error) {
+				select {
+				case <-continueDelete:
+					break
+				case <-ctx.Done():
+					return nil, nil
+				}
+				defer func() {
+					select {
+					case deleteFinished <- struct{}{}:
+						break
+					case <-ctx.Done():
+						break
+					}
+				}()
+				return h(ctx, deletionCriteria...)
+			}
+		}
+		deleteInterceptorProvider = &storagefakes.FakeDeleteInterceptorProvider{}
+		deleteInterceptorProvider.NameReturns("tetInterceptor")
+		deleteInterceptorProvider.ProvideReturns(deleteInterceptor)
+		defaultOlderThan = storage.DefaultSettings().Notification.OlderThan
+		testContext = common.NewTestContextBuilder().WithSMExtensions(func(ctx context.Context, smb *sm.ServiceManagerBuilder, e env.Environment) error {
+			repository = smb.Storage
+			smb.WithDeleteInterceptorProvider(types.NotificationType, deleteInterceptorProvider).Register()
+			return nil
+		}).WithEnvPreExtensions(func(set *pflag.FlagSet) {
+			err := set.Set("storage.notification.clean_interval", "0")
+			Expect(err).ToNot(HaveOccurred())
+		}).Build()
+	})
+
+	AfterSuite(func() {
+		testContext.Cleanup()
+	})
+
+	Context("When two notifications are inserted", func() {
+		It("Should delete the old one", func() {
+			idNew, err := repository.Create(ctx, randomNotification())
+			Expect(err).ToNot(HaveOccurred())
+			Expect(idNew).ToNot(BeEmpty())
+
+			oldNotification := randomNotification()
+			oldNotification.CreatedAt = time.Now().Add(-defaultOlderThan - time.Hour)
+			idOld, err := repository.Create(ctx, oldNotification)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(idOld).ToNot(BeNil())
+
+			getNotification(idOld)
+			getNotification(idNew)
+
+			continueDelete <- struct{}{}
```

Comment:
Can't this be simplified by setting the `older_than` configuration to be for example `30m` and leave the `created_at` of the old notification to be 1 hour ago?